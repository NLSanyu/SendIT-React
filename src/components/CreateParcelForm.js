import React, { Component } from "react";
import Button from "./Button";

export class CreateParcelForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: "",
      pickup_location: "",
      destination: ""
    };

  }


  render() {
    return (
      <div className="create-form-div">
        <h3>Create a new parcel</h3>
        <form className="create-form" id="create-parcel-form">
          <label htmlFor="desc">Description:</label>
          <input type="text" name="description" onChange={this.props.handleInputChange} />
          <label htmlFor="pickup">Pickup Location:</label>
          <input type="text" name="pickupLocation" onChange={this.props.handleInputChange} />
          <label htmlFor="dest">Destination:</label>
          <input type="text" name="destination" onChange={this.props.handleInputChange} />
          <br />
          <Button
            type="button"
            className="create-btn"
            id="create_parcel_btn"
            text="Create parcel"
            onClick={this.props.createParcel}
          />
        </form>
      </div>
    );
  }
}

export default CreateParcelForm;
