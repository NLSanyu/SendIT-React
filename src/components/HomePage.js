import React from "react";
import Header from "./Header";

export const HomePage = () => {
  return (
    <div>
      <Header />
      <div className="home-image">
        <img className="home-image" src="../assets/parcel.jpg" />
      </div>
      <div className="home-info">
        <p className="info-title">
          SendIT is a courier service that helps users deliver parcels to
          different destinations.
        </p>
        <p className="info-title">
          SendIT provides courier quotes based on weight categories
        </p>
        <p className="info-title">
          Benefit from the best services on the market, with our quick and safe
          deliveries and unbeatable prices. Get value for your money every
          single time!
        </p>
        <p className="info-title">
          Orders are priced starting from a base price of UGX (Uganda Shillings)
          3000
        </p>
      </div>
    </div>
  );
};

export default HomePage;
