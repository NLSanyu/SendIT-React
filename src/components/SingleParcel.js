import React from "react";

const SingleParcel = (props) => {
  return (
    <div className="profile-details">
      <p>Single Parcel</p>
    </div>
  );
};

export default SingleParcel;
