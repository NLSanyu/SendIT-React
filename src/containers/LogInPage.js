import React, { Component } from "react";
import { connect } from "react-redux";
import { logIn } from "../actions/LoginAction";
import Loader from "../components/Loader";
import Button from "../components/Button";
export class LogInPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };

    this.handleLogIn = this.handleLogIn.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleLogIn = () => {
    const userInfo = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.logIn(userInfo);
  };

  render() {
    return (
      <div className="auth">
        <h1>Log In</h1>
        <form name="log_in_form" id="log_in_form">
          <table className="form-table">
            <tbody>
              <tr>
                <td>Username</td>
                <td className="colon">:</td>
                <td>
                  {" "}
                  <input
                    name="username"
                    id="uname1"
                    type="text"
                    onChange={this.handleInputChange}
                    required
                  />{" "}
                </td>
              </tr>
              <tr>
                <td>Password</td>
                <td className="colon">:</td>
                <td>
                  {" "}
                  <input
                    name="password"
                    id="password"
                    type="password"
                    onChange={this.handleInputChange}
                    required
                  />{" "}
                </td>
              </tr>
            </tbody>
          </table>
          <Button
            type="button"
            className="submit-button"
            id="sign_up_btn"
            text="Log In"
            onClick={this.handleLogIn}
          />
        </form>
        <Loader />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  loggedIn: state.loggedIn
});

const mapDispatchToProps = {
  logIn
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LogInPage);
