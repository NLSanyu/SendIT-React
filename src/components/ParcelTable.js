import React from "react";
import SingleParcel from "./SingleParcel";

const ParcelTable = props => {

  const parcels = props.parcels ? props.parcels : [];
  const parcelItems = parcels.map(parcel => (
    <SingleParcel
      key={parcel.id}
      id={parcel.id}
      dateCreated={parcel.date_created}
      description={parcel.description}
      pickupLocation={parcel.pickup_location}
      destination={parcel.destination}
      price="UGX 3000"
      status={parcel.status}
    />
  ));

  return (
    <div id="all-parcels" className="parcels_details">
      <h3>My Parcels</h3>

      <div className="parcels-div">
        <ul>{parcelItems}</ul>
      </div>
    </div>
  );

};

export default ParcelTable;
