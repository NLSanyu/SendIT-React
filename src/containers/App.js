import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import { createBrowserHistory } from 'history';
import '../styles/App.css';
import HomePage from '../components/HomePage';
import SignUpPage from './SignUpPage';
import LogInPage from './LogInPage';
import ProfilePage from './ProfilePage';
import { ToastContainer } from 'react-toastify';

class App extends Component {
  render() {
    return (
      <div>
      <Router history={createBrowserHistory}>
        <Switch>
          <Route
              exact path='/'
              component={HomePage}
          />
          <Route
              path='/signup'
              component={SignUpPage}
          />
          <Route
              path='/login'
              component={LogInPage}
          />
          <Route
              path='/profile'
              component={ProfilePage}
          />
        </Switch>
      </Router>
      <ToastContainer />
      </div>
    );
  }
}
export default App;