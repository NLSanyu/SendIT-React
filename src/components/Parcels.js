import React, { Component } from "react";
import { connect } from "react-redux";
import CreateParcelForm from "./CreateParcelForm";
import ParcelTable from "../components/ParcelTable";
import { createNewParcel } from "../actions/ParcelActions";

export class Parcels extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: "",
      pickupLocation: "",
      destination: ""
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.createParcel = this.createParcel.bind(this);
    this.getParcels = this.getParcels.bind(this);
  }

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  createParcel = event => {
    const parcelInfo = {
      description: this.state.description,
      pickup_location: this.state.pickupLocation,
      destination: this.state.destination
    };
    this.props.createNewParcel(parcelInfo);
  };

  getParcels = () => {
    // alert("getParcels");
  };

  render() {
    return (
      <div className="parcels-details" id="all-parcels">
        <CreateParcelForm
          handleInputChange={this.handleInputChange}
          createParcel={this.createParcel}
        />
        <ParcelTable parcels={this.props.parcels} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  parcels: state.parcels
});

const mapDispatchToProps = {
  createNewParcel
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Parcels);
