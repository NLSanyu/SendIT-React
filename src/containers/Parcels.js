import React, { Component } from "react";
import { connect } from "react-redux";
import CreateParcelForm from "../components/CreateParcelForm";
import ParcelTable from "../components/ParcelTable";
import { createNewParcel, getAllParcels } from "../actions/ParcelActions";

export class Parcels extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: "",
      pickupLocation: "",
      destination: ""
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.createParcel = this.createParcel.bind(this);
  }

  componentDidMount() {
    setTimeout(() => {
      this.props.getAllParcels(this.props.user.user_id);
    }, 2000);
  }

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  createParcel = () => {
    const parcelInfo = {
      description: this.state.description,
      pickup_location: this.state.pickupLocation,
      destination: this.state.destination
    };
    this.props.createNewParcel(parcelInfo);
  };

  render() {
    return (
      <div className="parcels-details" id="all-parcels">
        <CreateParcelForm
          handleInputChange={this.handleInputChange}
          createParcel={this.createParcel}
        />
        <ParcelTable parcels={this.props.parcels} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  parcels: state.parcels.parcels,
  user: state.logIn.user
});

const mapDispatchToProps = {
  createNewParcel,
  getAllParcels
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Parcels);
