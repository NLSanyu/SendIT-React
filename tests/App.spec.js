import React from "react";
import { mount, shallow } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import App from "../src/containers/App";
import SignUpPage from "../src/containers/SignUpPage";
import LogInPage from "../src/containers/LogInPage";
import ProfilePage from "../src/containers/ProfilePage";
import HomePage from "../src/components/HomePage";
import Loader from "../src/components/Loader";
import CreateParcelForm from "../src/components/CreateParcelForm";
import ParcelTable from "../src/components/ParcelTable";
import SingleParcel from "../src/components/SingleParcel";
import Parcels from "../src/containers/Parcels";

describe("App", () => {
  const initialState = {
    signUp: { signedUp: false },
    logIn: { user: [] },
    parcels: { parcelCreated: false, parcels: [] }
  };
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it("app should render without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it("home page should render without crashing", () => {
    const wrapper = mount(
      <Router>
        <HomePage />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("sign up page should render without crashing", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <SignUpPage />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find("input#uname1")
      .first()
      .simulate("change");
  });

  it("log in page should render without crashing", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <LogInPage />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find("input#uname1")
      .first()
      .simulate("change");
  });

  it("profile page should render without crashing", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <ProfilePage />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("parcels page should render without crashing", () => {
    const wrapper = mount(<Provider store={store}>
      <Router>
        <Parcels />
      </Router>
    </Provider>);
    wrapper
    .find("input")
    .first()
    .simulate("change");
    wrapper
    .find("button")
    .first()
    .simulate("click");
  });

  it("single parcel form should render without crashing", () => {
    mount(<SingleParcel />);
  });

  it("loader should render without crashing", () => {
    mount(<Loader />);
  });
});
