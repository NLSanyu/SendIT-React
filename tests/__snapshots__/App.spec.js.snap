// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app should render without crashing 1`] = `ShallowWrapper {}`;

exports[`App home page should render without crashing 1`] = `ShallowWrapper {}`;

exports[`App log in page should render without crashing 1`] = `ReactWrapper {}`;

exports[`App profile page should render without crashing 1`] = `ReactWrapper {}`;

exports[`App should render header without crashing 1`] = `ShallowWrapper {}`;

exports[`App should render nav bar without crashing 1`] = `ShallowWrapper {}`;

exports[`App sign up page should render without crashing 1`] = `ReactWrapper {}`;
